integration/presto/src/main/prestodb/org/apache/carbondata/presto/CarbondataSplitManager.java

```diff
@@ -117,6 +122,16 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHand
       // file metastore case tablePath can be null, so get from location
       location = table.getStorage().getLocation();
     }
+    List<PartitionSpec> filteredPartitions = new ArrayList<>();
+    if (layout.getPartitionColumns().size() > 0 && layout.getPartitions().isPresent()) {
+      List<String> colNames =
+          layout.getPartitionColumns().stream().map(x -> ((HiveColumnHandle) x).getName())
+              .collect(Collectors.toList());
+      for (HivePartition partition : layout.getPartitions().get()) {
+        filteredPartitions.add(new PartitionSpec(colNames,
```

Comment:
according to this code all the PartitionSpec will have all the column names!!! is this correct?
The column names in partitionSpec should be per partition right?

Reply:
you mean all the partition column right ? not the all table columns ?

Ideally they use partitionPath. I have tested it. It didn't impacted. Let me check again.

Reply:
yeah all partition column, 1 partition spec should have only the columns it is created on, not all columns.

Reply:
I think HiveTableHandle will give matching column for the partitions only , because this is per query derived.

Reply:
I have verified. This scenario can work. I guess. No need to modify any code.

Reply:
```
presto:redods> select dtm,hh from dw_log_ubt_partition_carbon_neww;
     dtm     |     hh
-------------+------------
 part_dtm_01 | part_hh_01
 part_dtm_01 | part_hh_01
 part_dtm_01 | part_hh_02
 part_dtm_20 | part_hh_21
 part_dtm_01 | part_hh_03
 part_dtm_21 | NULL
(6 rows)

Query 20200910_035416_00017_wv9qh, FINISHED, 3 nodes
Splits: 22 total, 22 done (100.00%)
0:01 [6 rows, 176B] [9 rows/s, 282B/s]

presto:redods> select dtm,hh from dw_log_ubt_partition_carbon_neww where (dtm = 'part_dtm_01' and hh = 'part_hh_03') or dtm='part_dtm_21';
     dtm     |     hh
-------------+------------
 part_dtm_01 | part_hh_03
 part_dtm_21 | NULL
(2 rows)

Query 20200910_035548_00018_wv9qh, FINISHED, 3 nodes
Splits: 18 total, 18 done (100.00%)
0:01 [2 rows, 0B] [1 rows/s, 0B/s]


```